### ISSUE TITLE
Android app is slow with folder with large amount of entries 

### ISSUE THREAD

**Issue by BEST8OY** [floccusaddon/floccus#1750]
### Which version of floccus are you using?

5.3.1

### How many bookmarks do you have, roughly?

1800

### Are you using other means to sync bookmarks in parallel to floccus?

No

### Sync method

Google Drive

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Edge 130, Firefox 131, Android app 5.3.1

### Describe the Bug

I have a folder containing 1800 entries and it makes the Android app really slow!

### Expected Behavior

I want it to be speedy and fluent (as firefox bookmark manager and chromium based ones because those are slow)

### To Reproduce

Have a large amount of entries in a folder and try opening it in the app.
If you try to have a large amount of entries in the main folder the app even struggles to open.

### Debug log provided

- [ ] I have provided a debug log file

**Comment by cyrusyamir** [comment]
i am using git for sync but not able to sync to mobile app, sometimes app is also get freeze. about 400 links (bookmarks), single browswe (Edge)

**Comment by BEST8OY** [comment]
> i am using git for sync but not able to sync to mobile app, sometimes app is also get freeze. about 400 links (bookmarks), single browswe (Edge)

Yeah.
The app has problem with GIT
I had to switch to Google drive

**Comment by cyrusyamir** [comment]
@BEST8OY Does Gdrive sync work two way or on mobile it shows sync bookmarks, links.

**Comment by BEST8OY** [comment]
> @BEST8OY Does Gdrive sync work two way or on mobile it shows sync bookmarks, links.

Two-way
You can add items in the app and sync

**Comment by marcelklehr** [comment]
> I have a folder containing 1800 entries and and when I'm opening the folder the Android app becomes really slow!

Yep, this is a known issue. It's not easy to fix, sadly.

**Comment by daffydock** [comment]
> > I have a folder containing 1800 entries and and when I'm opening the folder the Android app becomes really slow!
> 
> Yep, this is a known issue. It's not easy to fix, sadly.

Asking just for my knowledge. Does Fluccus search the server list or does it keep a copy locally on the app on Android? I thought it keep a copy locally. 

Since I really like your app but I am also worried of saving/putting too many bookmarks on the bookmark app as I concern with the fact that it becomes sluggish and then by extension Floccus. Great app name, by the way.

I made the mistake of going hog wild on the NC Notes app only for the Notes App Devs to later admit that the Notes app can't really handle thousands of notes before it starts crapping out. So now, I have odd sync issues with it. 

As I noticed that in another issue you mentioned of a test account with 18k bookmarks. Which seemed kind of impressive.

**Comment by marcelklehr** [comment]
> Does Fluccus search the server list or does it keep a copy locally on the app on Android? I thought it keep a copy locally.

Yes, it keeps a local copy in the Android/iOS apps as well, just like in the browser. As long as you don't have all your 18k bookmarks in a single folder, but nicely spread across folders, the app should be ok, I think.
